jax/test_util.py

```diff
@@ -246,6 +247,47 @@ def f_vjp(*args):
 
   _check_grads(f, args, order)
 
+
+@contextmanager
+def count_primitive_compiles():
+  xla.xla_primitive_callable.cache_clear()
+
+  # We count how many times we call primitive_computation (which is called
+  # inside xla_primitive_callable) instead of xla_primitive_callable so we don't
+  # count cache hits.
+  primitive_computation = xla.primitive_computation
+  count = [0]
+
+  def primitive_computation_and_count(*args, **kwargs):
+    count[0] += 1
+    return primitive_computation(*args, **kwargs)
+
+  xla.primitive_computation = primitive_computation_and_count
+  try:
+    yield count
```

Comment:
Maybe slightly cleaner
```suggestion
    yield count[0]
```

Reply:
We have to yield a list (or some kind of object reference) so that when we update it inside the context manager, it's reflecting in the test function as well (we basically need an int*). I could probably make some little wrapper thingy but this seems simpler, if uglier in test code.